atd-vze/src/views/Tools/UnitsForm.js

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+
+import {
+  Card,
+  CardBody,
+  CardHeader,
+  Col,
+  Input,
+  Label,
+  FormGroup,
+  FormText,
+} from "reactstrap";
+import { useQuery } from "@apollo/react-hooks";
+import { withApollo } from "react-apollo";
+import { gql } from "apollo-boost";
+
+const GET_VEHICLE_DESC_LKP = gql`
+  {
+    atd_txdot__veh_unit_desc_lkp {
+      veh_unit_desc_id
+      veh_unit_desc_desc
+    }
+  }
+`;
+
+const UnitsForm = ({ units, handleUnitFormChange, client }) => {
+  const { data: lookupValues, loading } = useQuery(GET_VEHICLE_DESC_LKP);
+
+  return (
+    <>
+      {units.map((unit, i) => {
+        return (
+          <Card key={`unitCard-${i + 1}`}>
+            <CardHeader>Unit #{i + 1}</CardHeader>
+            <CardBody>
+              <FormGroup row>
+                <Col md="3">
+                  <Label htmlFor="unit-type">Unit Type</Label>
+                </Col>
+                <Col xs="12" md="9">
+                  <Input
+                    type="select"
+                    id="unit-type"
+                    name="unit-type"
+                    placeholder="Enter Fatality Count..."
+                    value={unit.unit_desc_id}
+                    onChange={e =>
+                      handleUnitFormChange({
+                        type: "unit_desc_id",
+                        payload: e.target.value,
+                        unitIndex: i,
+                      })
```

Comment:
Basically `handleUnitFormChange` is an alias for `unitFormDispatch`, here we pass the type of event, the data (payload), and the array position (index) at which the state needs to be updated. 👍

Reply:
yup, its pretty similar in theory to Redux actions and dispatch.

It's a convoluted metaphor but I like the cartoons here:
https://code-cartoons.com/a-cartoon-intro-to-redux-3afb775501a6

![image](https://user-images.githubusercontent.com/5697474/89970532-a561c880-dc1e-11ea-9452-63fff289e12b.png)


Another random image:
![image](https://user-images.githubusercontent.com/5697474/89970579-bca0b600-dc1e-11ea-92b1-d736a3f91b56.png)